### PR TITLE
feat: Add functions to queue  Remove, Search, and SearchAll package jobs

### PR DESCRIPTION
**Pull request by sindharta**

### CODE REVIEW

File: Packages/com.unity.renderstreaming/Editor/RequestJobManager.cs [hunk 1]
```diff
@@ -80,6 +131,36 @@ static void UpdateRequestJobs()
...
+                m_requestJobs.Add(new RequestJob(removeReq,info.OnSuccessAction,info.OnFailAction));
+            }
+            m_pendingRemoveRequests.Clear();
+        }
+
+        {   //Process pending SearchRequests
+            var enumerator = m_pendingSearchRequests.GetEnumerator();
+            while (enumerator.MoveNext()) {
+                SearchRequestInfo info = enumerator.Current;
+                SearchRequest removeReq = Client.Search(info.PackageName, info.OfflineMode);
```
Comment (karasusan) [inline Packages/com.unity.renderstreaming/Editor/RequestJobManager.cs]: removeReq?

File: Packages/com.unity.renderstreaming/Editor/RequestJobManager.cs [hunk 2]
```diff
@@ -80,6 +131,36 @@ static void UpdateRequestJobs()
...
+                m_requestJobs.Add(new RequestJob<PackageInfo[]>(removeReq,info.OnSuccessAction,info.OnFailAction));
+            }
+            m_pendingSearchRequests.Clear();
+        }
+
+        {   //Process pending SearchAllRequests
+            var enumerator = m_pendingSearchAllRequests.GetEnumerator();
+            while (enumerator.MoveNext()) {
+                SearchAllRequestInfo info = enumerator.Current;
+                SearchRequest removeReq = Client.SearchAll(info.OfflineMode);
```
Comment (karasusan) [inline Packages/com.unity.renderstreaming/Editor/RequestJobManager.cs]: removeReq?

File: Packages/com.unity.renderstreaming/Editor/RequestJobManager.cs (hunk 1 above)
Comment (sindharta) [reply Packages/com.unity.renderstreaming/Editor/RequestJobManager.cs]: thanks. will fix

File: Packages/com.unity.renderstreaming/Editor/RequestJobManager.cs (hunk 2 above)
Comment (sindharta) [reply Packages/com.unity.renderstreaming/Editor/RequestJobManager.cs]: thanks. will fix

**Pull request merged**